### PR TITLE
7 am jp no items message in list component

### LINKED ISSUE #8: 7. Como usario, quiero ver un mensaje emergente para añadir mi primer item a la lista si esta está vacía, así ayudarme a orientarme sobre cómo funciona la app
AC:

* La vista de la lista, cuando no tiene items para mostrar, debe mostrar un mensaje (ej. un botón) para que el usuario pueda agregar un nuevo item

### PR DESCRIPTION
**Pull request by dupjpr**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

In this PR we create a new message in the list component, which will be displayed when a list is empty. In this message there is information about what the user should do in case there is no item in the list. 

We did it using the firebase object response, where there is a key with the name of 'empty'. This key is false when the document has no elements and it is true when there is some element inside.

Finally we create an additional validation in the render of the list component, because the element {ul} is displayed on the screen even when there are no elements in the list. To correct this error, we create a double validation using the 'empty' key from firebase's response.

## Related Issue

closes #8 

## Acceptance Criteria

- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|   ✓| :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Captura de Pantalla 2021-04-27 a la(s) 5 24 06 p  m

### After

Captura de Pantalla 2021-04-27 a la(s) 5 22 36 p  m

## Testing Steps / QA Criteria

From your terminal, pull down the branch
Run npm install to install the dependencies
Run npm start to view the features in the browser
Created a new list to watch the landing message

### CODE REVIEW

File: src/components/List/List.jsx [hunk 1]
```diff
@@ -15,7 +15,15 @@ const List = () => {
     <div>
       {error && <strong>Error: {JSON.stringify(error)}</strong>}
       {loading && <span>Collection: Loading...</span>}
-      {value && (
+
+      {value && value.empty && (
+        <p>
+          There are not items in this list yet. You can add items with the "Add
```
Comment (MikeBLambert) [inline src/components/List/List.jsx]: Super tiny grammatical fix: "There are not _any_ items..." or "There are _no_ items..."

File: src/components/List/List.jsx [hunk 2]
```diff
@@ -15,7 +15,15 @@ const List = () => {
     <div>
       {error && <strong>Error: {JSON.stringify(error)}</strong>}
       {loading && <span>Collection: Loading...</span>}
-      {value && (
+
+      {value && value.empty && (
```
Comment (MikeBLambert) [inline src/components/List/List.jsx]: Nit: we can use optional chaining rather than the `&&`

**Review by MikeBLambert — APPROVED** [review]
Nice Job!

**Review by candrepa1 — APPROVED** [review]

File: src/components/List/List.jsx (hunk 2 above)
Comment (FauricioValencia) [inline src/components/List/List.jsx]: what do you think if use something like this:

`value?.empty && (`

File: src/components/List/List.jsx [hunk 3]
```diff
@@ -15,7 +15,15 @@ const List = () => {
...
+        <p>
+          There are not items in this list yet. You can add items with the "Add
+          Item" button at the bottom!
+        </p>
+      )}
+
+      {value && !value.empty && (
```
Comment (FauricioValencia) [inline src/components/List/List.jsx:45]: what do you think if use something like this:
`!value?.empty &&`

**Review by FauricioValencia — APPROVED** [review]

File: src/components/List/List.jsx (hunk 2 above)
Comment (dupjpr) [reply src/components/List/List.jsx]: Thanks, it has been solved. 👍🏻

Comment (dupjpr) [reply src/components/List/List.jsx]: Thanks, it has been solved. 👍🏻

**Pull request merged**